src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -811,7 +811,7 @@ class PersonAgent(
           if (currentVehicle.isEmpty || currentVehicle.head != nextLeg.beamVehicleId) {
             beamVehicles(nextLeg.beamVehicleId) match {
               case t @ Token(_, manager, _) =>
-                manager ! TryToBoardVehicle(t, self)
+                manager ! TryToBoardVehicle(t, self, getCurrentTriggerId.getOrElse(-1111))
```

Comment:
This gives me pause. Should we be using triggerId for what seems to be correlationId? While this has good intentions, it could lead to more work down the road, and mis-diagnosis - for the case especially where an Id does not exist - and then is propagated as though it DOES exist. Rather then trying to piggyback and twist on something that already has a purpose, we should use a means that is specific to what is needed. For example, use a correlationId instead - or utilize https://opentelemetry.io/, which is becoming more of a standard for this type of thing.

Reply:
Replaced with getCurrentTriggerIdOrGenerate